doc/radosgw/lua-scripting.rst

```diff
@@ -0,0 +1,338 @@
+=============
+Lua Scripting
+=============
+
+.. versionadded:: Pacific
+
+.. contents::
+
+This feature allows users to provide a Lua script that will be executed after a request has been processed.
+The script will have access to fields in the request, and may also modify some fields. All Lua language features could be used in the script.
+
+
+.. toctree::
+   :maxdepth: 1
+
+
+Script Management via CLI
+-------------------------
+
+Each user may have one script which is executed in the context of the S3 requests of that user.
+
+To upload a script use:
+   
+::
+   
+   # radosgw-admin script put --infile={lua-file} --uid={user-id}
+
+
+To print the content of the script to standard output:
+
+::
+   
+   # radosgw-admin script get --uid={user-id}
+
+
+To remove the script:
+
+::
+   
+   # radosgw-admin script rm --uid={user-id}
+
+
+Context Free Functions
+----------------------
+Debug Log
+~~~~~~~~~
+The ``RGWDebugLog()`` function accepts a string message and print it to the debug log with priority 20. 
+Each message will be prefixed with: ``Lua INFO:``. This function has no return value.
+
+
+S3 Request Fields
+-----------------
+
+.. note::
+
+    - As this feature is experimental, fields may be removed and field names may be changed in the future
+    - Even tough Lua is case sensitive language, all field names provided by the radosgw to Lua are case insensative. Function names remain case sensitive
```

Comment:
Although Lua is a case-sensitive language, field names provided by the radosgw are case-insensitive. Function names remain case-sensitive.